### PR TITLE
Feature: required columns, column alias binding

### PR DESCRIPTION
**Pull request by alex-haulmont**
- removed strict ordering headers check for template matching
- added flag to mark is column required
- corrected file validation for related case
- corrected tests

### CODE REVIEW

File: modules/global/src/de/diedavids/cuba/dataimport/entity/attributemapper/messages.properties [hunk 1]
```diff
@@ -10,6 +10,7 @@ ImportAttributeMapper.attributeType = Attribute type
 ImportAttributeMapper.associationLookupAttribute = Association lookup attribute
 ImportAttributeMapper.mapperMode = Mode
 
+ImportAttributeMapper.isRequiredColumn=Is required column
```
Comment (mariodavid) [inline modules/global/src/de/diedavids/cuba/dataimport/entity/attributemapper/messages.properties:13]:
```suggestion
ImportAttributeMapper.isRequiredColumn=Column Required
```

File: modules/web/src/de/diedavids/cuba/dataimport/web/importfile/ImportDataImportConfigurationMatchException.groovy [hunk 2]
```diff
@@ -1,8 +1,11 @@
 package de.diedavids.cuba.dataimport.web.importfile
 
+import de.diedavids.cuba.dataimport.dto.ColumnValidationResult
+
 class ImportDataImportConfigurationMatchException extends RuntimeException {
 
-    ImportDataImportConfigurationMatchException() {
-        super('Import File does not match import configuration')
+    ImportDataImportConfigurationMatchException(ColumnValidationResult result) {
+        super('Import File does not match import configuration. Please check required columns: '
```
Comment (mariodavid) [inline modules/web/src/de/diedavids/cuba/dataimport/web/importfile/ImportDataImportConfigurationMatchException.groovy:8]: I would store the `ColumnValidationResult` variable here as an attribute in order to use it in the frontend to have a translation and pass those fields in as a parameter.

File: modules/web/src/de/diedavids/cuba/dataimport/web/importwizard/ImportWithImportConfigurationWizard.groovy [hunk 3]
```diff
@@ -159,7 +159,7 @@ class ImportWithImportConfigurationWizard extends AbstractEditor<ImportConfigura
             log.error(e.message, e)
 
             startImport.enabled = false
-            showNotification(formatMessage('uploadFileDoesNotMatchExpectedStructure'), Frame.NotificationType.ERROR)
```
Comment (mariodavid) [inline modules/web/src/de/diedavids/cuba/dataimport/web/importwizard/ImportWithImportConfigurationWizard.groovy:162]: note sure about this change. This way other people cannot define translations for them, right?

Perhaps alternatively you can you like `formatMessage('uploadFileDoesNotMatchExpectedStructure', e.getResult())` where `getResult` is the method of the exception which contains the validation result.

Comment (alex-haulmont) [reply modules/web/src/de/diedavids/cuba/dataimport/web/importwizard/ImportWithImportConfigurationWizard.groovy:162]: I generated scripts using Cuba features, installed docker images from mysql and postgress as you suggested at create-databases.md.
Translation changes are coming soon

**Pull request merged**